src/hooks/useCheckout.js

```diff
@@ -0,0 +1,23 @@
+import { onAuthStateChanged } from 'firebase/auth'
+import { useDispatch, useSelector } from 'react-redux'
+import { auth } from '../firebase'
+import { useEffect, useState } from 'react'
+import { login, logout } from '../store/slices/auth'
+
+export const useCheckout = () => {
+   const [isLoading, setIsLoading] = useState(true)
+   const { status } = useSelector((state) => state.auth)
+   const dispatch = useDispatch()
+
+   useEffect(() => {
+      onAuthStateChanged(auth, (user) => {
+         setIsLoading(false)
```

Comment:
Even if this solution is simple and it works, I would like you to think if there is a 'loading' state already create, and indeed it is, we have a 'checking', can we use that instead?